### DISCUSSION TITLE
1080 gif creation

### DISCUSSION THREAD

**Discussion by nembdev** [ImageMagick/ImageMagick#4765] [Help] [answered]
Hello,
I'm trying to create a gif from images.
I'm using python and ImageMagick.
My test run had

- 49 pngs @ 60.1mb

- 1920x1080p <- where I think my issue is

Using ImageMagick's convert function:
A single cpu core was maxed at 100%, disk write was climbing to 7+gbs.
The gif never completed after about 20 minutes, but I did get a partially finished one @ 40mb in size.

Is making 1080 gifs that resource intensive? Is there a better way? Thanks.

```
def create_gif(inputPath, outputPath, delay, finalDelay, loop):
    # grab all image paths in the input directory
    imagePaths = sorted(list(paths.list_images(inputPath)))

    # remove the last image path in the list
    lastPath = imagePaths[-1]
    imagePaths = imagePaths[:-1]
    
    # construct the image magick 'convert' command that will be used
    # generate our output GIF, giving a larger delay to the final
    # frame (if so desired)
    cmd = "convert -delay {} {} -delay {} {} -loop {} {}".format(
        delay, " ".join(imagePaths), finalDelay, lastPath, loop, outputPath
    )
    os.system(cmd)


create_gif(path, out_path, 0, 0, 0)
```

**Comment by fmw42** [comment, marked as answer]
1920x1080p by 49 might tax your RAM. ImageMagick needs to load them all at once plus the output.  You can use disk space in place of RAM by adjusting your resources. See policy.xml and 

`convert -list resource`

**Reply by nembdev** [reply]
Thanks for the response.
I just checked and my 8gb was 77% utilized. 
I had already changed the disk 1gb limit to 4gb I believe. 
```
<policy domain="resource" name="disk" value="4GiB"/>
```
Would more ram increase the conversion speed?
I assume these writes would be detrimental to a ssd?
Thanks.

Edit:
I have now changed:
```
  <!--<policy domain="resource" name="area" value="128MP"/>-->
  <policy domain="resource" name="area" value="5GiB"/>
```
Testing this now.

**Reply by fmw42** [reply]
How much disk space do your input images use?  Do you have that much for the final output?

**Reply by nembdev** [reply]
I adjusted the memory as well.
That seems to have done it. My first gif completed under 5 minutes.

My original input was 60mb in size total. Resulted in a partial output of 40mb.
I halved my input images to 32.2mb for testing, and this resulted in 23.3mb output.

The memory value was key. Thanks Again.

**Comment by snibgo** [comment]
The units for area is P for Pixels, not B for bytes. The area of an image is the width multplied by the height, in pixels.

I think the "area" resource is per image, but perhaps it is a total across all images.

Your operating system can tell you if the temporary location on disk is used while your command is running. Disk is slower than RAM.

Your roughly 50 input files are roughly 2 MP each, so 100 MP total. Depending on your IM, this might need 8 to 24 bytes per pixel, so up to 2.4 GB, before whatever memory is needed to stick images together into a GIF.

**Reply by nembdev** [reply]
Thank you for your response as well.
I've corrected the area and learned something lol.
